register.html

```diff
@@ -75,24 +79,37 @@ <h3><br/>Add a new entry</h3>
         <div class="form-group">
             <label class="control-label col-sm-2" for="address">Spot Address:</label>
             <div class="col-sm-10">
-                <input type="text" class="form-control" id="address"
-                       placeholder="Enter the full address as seen in Google Maps" name="address">
+                <input type="text" class="form-control" id="address" onfocus="geolocate()"
+                       placeholder="Enter the full address" name="address">
+            </div>
+        </div>
+
+        <div class="form-group hidden">
+            <label class="control-label col-sm-2" for="formatted-address">Formatted address:</label>
+            <div class="col-sm-10">
+                <input type="text" class="form-control" id="formatted-address" name="formatted-address" readonly>
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label class="control-label col-sm-2" for="get-lat-long"></label>
+            <div class="col-sm-4">
+                <input type="button" class="btn btn-default" id="get-lat-long" onclick="getCoordinates()"
+                       name="get-lat-long" value="Get GPS Coordinates">
             </div>
         </div>
 
         <div class="form-group">
             <label class="control-label col-sm-2" for="latitude">Latitude:</label>
             <div class="col-sm-10">
-                <input type="text" class="form-control" id="latitude"
-                       placeholder="Copy paste the latitude coordinate from Google Maps" name="latitude">
+                <input type="text" class="form-control" id="latitude" name="latitude" readonly>
             </div>
         </div>
 
         <div class="form-group">
             <label class="control-label col-sm-2" for="longitude">Longitude:</label>
             <div class="col-sm-10">
-                <input type="text" class="form-control" id="longitude"
-                       placeholder="Copy paste the longitude coordinate from Google Maps" name="longitude">
+                <input type="text" class="form-control" id="longitude" name="longitude" readonly>
```

Comment:
I like the fact that these fields are `readonly` . That way the user would only be able to generate latitude and longitude values by clicking the corresponding button.